packages/react-components/src/components/PromoBanner/PromoBanner.tsx

```diff
@@ -32,23 +34,51 @@ export const PromoBanner: React.FC<PromoBannerProps> = ({
   img,
   light = false,
   linkText,
-  size = 'small',
   onButtonClick,
   onClose,
   onLinkClick,
 }) => {
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const [containerSize, setContainerSize] = React.useState<
+    'small' | 'medium' | 'large'
+  >('medium');
+
   const mergedClassNames = cx(
     styles[baseClass],
     {
       [styles[`${baseClass}--light`]]: light,
-      [styles[`${baseClass}--${size}`]]: size,
+      [styles[`${baseClass}--small`]]: containerSize === 'small',
+      [styles[`${baseClass}--large`]]: containerSize === 'large',
     },
     className
   );
 
-  const shouldRenderLargeFooter = (buttonText || linkText) && size === 'large';
-  const shouldRenderSmallOrMediumFooter =
-    (buttonText || linkText) && size !== 'large';
+  React.useEffect(() => {
+    const handleResize = debounce(() => {
+      if (
+        containerRef.current &&
+        containerRef.current.offsetWidth <= SMALL_CONTAINER_WIDTH_TRESHOLD
+      ) {
+        return setContainerSize('small');
+      }
+
+      if (
+        containerRef.current &&
+        containerRef.current.offsetWidth >= LARGE_CONTAINER_WIDTH_TRESHOLD
+      ) {
+        return setContainerSize('large');
+      }
+
+      return setContainerSize('medium');
+    }, RESIZE_DEBOUNCE_TRESHOLD);
+
+    window.addEventListener('resize', handleResize);
```

Comment:
While finally testing your changes, I have spotted a bug related to the initial size of the component when the window is initially small or large. Here is a proposal for fixing this behavior.

In addition to registering the `handleResize` as a window resize event handler, it should be immediately called after the component is mounted to adjust it to the initial window size. Currently, the component is stuck with `'medium'` size until the first window resizes happens, but it can be loaded into large or small sizes at the beginning. 

To do this properly, you don't want to call the debounced function not to introduce unnecessary delay on the init. I would propose extracting the function denounced here to its own variable and calling it directly. It can be done like this:

```ts
    const handleResize = () => {
       /* ... */
    }

    const debouncedHandleResize = debounce(handleResize, RESIZE_DEBOUNCE_TRESHOLD);
    window.addEventListener('resize', debouncedHandleResize);

    handleResize();

    return () => {
      debouncedHandleResize.cancel();
      window.removeEventListener('resize', debouncedHandleResize);
    };
```

Reply:
👍